cime_config/usermods_dirs/NEON/YELL/shell_commands

```diff
@@ -3,14 +3,12 @@
 ./xmlchange PTS_LON=249.45803999999998
 ./xmlchange PTS_LAT=44.95597
 ./xmlchange DATM_YR_ALIGN=2019
-# NEON precipitation data for this site is bad so end early
+# NEON precipitation data for this site is missing so end early
 # If CLM_USRDAT_NAME is NEON.PRISM you can run to the end of the data
-if [ $CLM_USRDAT_NAME=='NEON' ]
-then
+if [[ $CLM_USRDAT_NAME == 'NEON' ]]; then
   # Different default start date and number of months to run for transient case
   ./xmlchange DATM_YR_START=2019
-  if  [[ $compset =~ ^HIST ]]
-  then
+  if  [[ $compset =~ ^HIST ]]; then
     ./xmlchange RUN_STARTDATE=2019-01-01
     ./xmlchange STOP_N=39
```

Comment:
This applies to all the user-mods, but explicitly setting STOP_N may interfere with the users own setting, test-mods, or run_neon.py --end-date option.

Reply:
We also have STOP* changes in cime_config/usermods_dirs/NEON/defaults/shell_commands, which then get overwritten by these site-specific xmlchanges.

Is there a way to set the STOP variables to only do an xmlchange IF there is not already an xmlchange (or two) for STOP variables listed in the case status, perhaps?

Reply:
It would be awkward to check if there already is an xmlchange for it. I think it's possible, but I'd also view it as really fragile code that could be problematic.

But, that also wouldn't know about if the --end-date option was used in run_neon.py or if the user wanted to set it to something explicit.

Since, we have a time crunch here, I'm also OK with documenting it as a problem, and then making the tag. But, I also wanted to hear from everyone about if they think it will cause a problem for the tutorial users.  And if so we probably should do something. Although I think it would require that they "go away from the instruction" and do something wild.

Reply:
That certainly makes sense. Good point. I personally don't recall folks adjusting STOP variables in the previous tutorial, and wouldn't really expect users to need to do that in the flux course, but perhaps others can chime in.

Reply:
as far as I know the `--end-date` (and `--start-date`) options used in run_neon.py are not functional, we should likely just take this out to avoid confusion (but so far it hasn't been an issue).

Reply:
Thanks for the discussion. From #1906 it looks like --end-date (and --run-length) are functional. I think this warrants more discussion because a common thing for a user is to be the one that controls the length of the simulation. So putting this in the user-mod directory is problematic.

But, for right now to get a tag out -- I'm going to check on the XML variable "TEST" so I can ignore this for my tests. I'll file an issue about setting STOP options and we can also discuss it tomorrow.

Reply:
We'll discuss more in #2036